src/spyglass/decoding/v1/sorted_spikes.py

```diff
@@ -401,10 +409,126 @@ def load_spike_data(key, filter_by_interval=True):
         min_time, max_time = SortedSpikesDecodingV1._get_interval_range(key)
 
         new_spike_times = []
-        for elec_spike_times in zip(spike_times):
+        for elec_spike_times in spike_times:
             is_in_interval = np.logical_and(
                 elec_spike_times >= min_time, elec_spike_times <= max_time
             )
             new_spike_times.append(elec_spike_times[is_in_interval])
 
         return new_spike_times
+
+    @classmethod
+    def get_spike_indicator(cls, key, time):
```

Comment:
This is the same across `clusterless` and `sorted_spikes`, so I think it should be moved to being a util to avoid redundancy. It can still accept the relevant class as an arg, but perhaps renamed, or just the spike times

`from spyglass.decoding.v1.utils import get_spike_indicator`

```python
def get_spike_indicator(spike_times, time):
    ...
    return spike_indidator
```

Reply:
Yeah, due to time constraints, I haven't managed to refactor all of this, but agreed.